### ISSUE TITLE
Adopting vigiles in a skills monorepo: init adoption, testGlobs layout, `.claude/rules/`

### ISSUE THREAD

**Issue by vlad-ryzhkov** [zernie/agent-lint#175]
Tested on `vigiles@21.0.1` (npm) = `681edf7`.

Context: adopting vigiles in a repo with 51 skills, 4 plugins, no TypeScript, and a centralized test
layout. Three separate gaps showed up on that path. They are small and related, so one issue rather
than three.

---

## 1. `init --target=<CLAUDE.md>` yields a spec that verifies nothing

Took a real `CLAUDE.md` — 200 lines, a JIT routing table, ~40 path refs in backticks:

```
$ npx vigiles init --target=CLAUDE.md --yes --no-gha --no-plugin
Adopted CLAUDE.md → CLAUDE.md.spec.ts (raw, 1 section).
```

Result is the whole file in one template string, `rules: {}`, zero extracted refs:

```ts
// Adopted from CLAUDE.md by `vigiles init` — faithful by default.
// Each heading became a prose section; no rules were inferred.
export default instructionFile({
  sections: { "Overview": `<the entire file as one string, every backtick escaped>` },
  rules: {},
});
```

The cost lands immediately: the file becomes a build artifact, manual edits get blocked by the hook,
every backtick is escaped, editing now happens in TS. The benefit is zero until someone rewrites
every ref by hand. For a team deciding whether to move to spec mode, that reads as "expensive and
useless" — while the mechanism underneath works fine: a hand-written `file("docs/…")` is caught by
`compile` correctly.

**Suggested patch — don't touch the parser, add one output line.** Extracting refs by heuristic is
new code and a new false-positive source, the same one that got `doc-refs` disabled (`7b8cdec`). The
cheap version: after `Adopted … (raw, N sections)`, print a second line when `rules` is empty and no
refs were extracted:

```
  0 refs extracted — this spec verifies nothing yet. Wrap paths in file() and
  commands in cmd() to make compile check them.
```

Expectations match reality, no new false positives.

---

## 2. `testGlobs` cannot express a centralized `tests/<surface>/` layout

Our tests live in `tests/<skill>/evals/promptfooconfig.yaml`, not next to the skill. Configured the
documented workaround:

```json
{ "rules": { "untested-skill": ["warn", { "testGlobs": ["tests/*/evals/promptfooconfig*.yaml"] }] } }
```

The uncovered count did not move: **35 before, 35 after**. `testGlobs` widens the set of files that
count as tests (`src/test-coverage.ts:647`), but the colocation-and-name rule
(`skills/foo/foo.eval.mjs`) stays, and our files satisfy neither the place nor the name.

The rationale in `docs/rules/untested-skill.md` is that "`ls` answers 'is it covered?' without
running anything". In a centralized layout `ls tests/<skill>/` answers the same question just as
cheaply, only elsewhere. The rule is not forbidding an expensive lookup, it is forbidding a
different convention.

**Suggested patch — surface-name substitution in the glob.** Allow `{surface}` in `testGlobs`:

```json
{ "testGlobs": ["tests/{surface}/evals/promptfooconfig*.yaml"] }
```

Substituting before matching is one string replace where `testGlobs` is expanded into a matcher. The
mapping stays unambiguous (the name still says what the test is about); only the place becomes
configurable — the property the rule was introduced for.

If that is too much, the output hint at least needs fixing: it currently points at `testGlobs`, and
`testGlobs` does not solve this.

---

## 3. No surface for `.claude/rules/**` — a whole instruction layer is invisible

Claude Code loads `.claude/rules/*.md` as project instructions, path-scoped via `paths:` in the
frontmatter. In one session five such files from our repo arrived in context labelled "project
instructions".

`PluginLayout` (`src/core/layout.ts:13`) describes `instructionFile`, `skillDir`, `agentDir`,
`commandDir`, hooks, settings and MCP. There is no rules directory.

Checked it: built a fixture with `.claude/rules/git.md` containing `paths:` and a ref to a
nonexistent file. `vigiles lint` did not mention the file at all — not in the scanned list, not in
the findings.

Why this one matters here specifically: this is exactly the layer where "rules the agent follows
sometimes" live — the claim on the first screen of the README. Our hardest repo policies sit in
`.claude/rules/`, and to a harness audit they do not exist.

**Suggested patch — one optional field, no default behaviour change.** In `PluginLayout`, next to
`commandDir` (`src/core/layout.ts:60`):

```ts
  /** Path-scoped rules dir, holding flat `<dir>/<name>.md` (`""` = none). */
  readonly rulesDir?: string;
```

The Claude Code adapter fills it with `"rules"`, others leave it empty. Then include it in the same
list as `commandDir` in the surface scanner, so the files land in `frontmatter-valid` and in the
rule map. Nothing new to check — the existing checks just see one more directory.

**Comment by zernie** [comment]
All three shipped in #186 (`94af23a5`). Taking them in the order you filed them, because the answers differ.

## 1. `init --target=` — the output line, as you suggested

Implemented exactly as proposed: no heuristic ref extraction, one honest line. A raw adoption now prints

```
  ℹ 0 refs extracted — this spec verifies nothing yet. Wrap paths in `file()`
    and commands in `cmd()` to make `compile` check them.
```

Your reasoning for *not* touching the parser is the reason it was not touched — guessing refs out of prose is what got `doc-refs` disabled (measured at 0 true positives across 2 582 markdown files). The cost lands immediately and the benefit is not-yet-claimed rather than zero; saying so is the whole fix.

## 2. `testGlobs` — `{surface}` shipped, and your diagnosis was exactly right

`testGlobs` widens what counts as a *test file* and says nothing about which *surface* a file is for, which is why your uncovered count did not move. The placeholder is substituted with each skill's own name before matching:

```json
{ "rules": { "untested-skill": ["warn", { "testGlobs": ["tests/{surface}/evals/promptfooconfig*.yaml"] }] } }
```

A glob **without** `{surface}` still credits nothing, and that is deliberate rather than an oversight: inferring the surface from a substring is precisely the retired `name-mentioned` tier, which credited surfaces no test had touched.

Two things worth flagging, since you read the rationale carefully enough to argue with it:

- **The cost is real and is now written down in the rule doc.** `ls` beside the skill no longer answers "is this tested?" — that property is the argument colocation was chosen on. It is opt-in per repo for that reason, not a second default.
- **Your report exposed a latent bug on the way in.** `coverageOf` promised "STRONGEST evidence, not first-found" in its doc comment and actually kept the first match. Harmless while colocation was the only tier — one tier cannot be out-ranked — and load-bearing the moment a second existed, since the reported provenance would otherwise depend on the *order of your `testGlobs` array*. There is a real rank now, pinned by a test that runs both orders.

The output hint was also wrong, as you said — it pointed at `testGlobs` for a problem `testGlobs` did not solve. It now names both accepted shapes and says plainly that a glob without the placeholder credits nothing.

## 3. `.claude/rules/**` — `rulesDir` shipped, with one deliberate difference

Added as an optional `PluginLayout` field, Claude Code fills it with `"rules"`, and the files now reach `frontmatter-valid` and the rule map. Verified on a fixture: a rules file with unparseable YAML is reported by `lint` at `.claude/rules/git.md` where before it was not mentioned at all.

The difference from your patch: you suggested including it "in the same list as `commandDir` in the surface scanner". I read the files **outside** `surfaceDirs` instead. That list decides whether a directory counts as a *loadable machine*, and rules are instructions, not an invocable surface — folding them in would silently redefine what "an empty machine" means for every harness, including ones with no rules layer at all. So they are added to the file map that text checks read, and left out of the counts. A layout without `rulesDir` behaves byte-for-byte as before, which is tested.

One thing that only showed up in implementation: keying the read off the resolved scopes reads the **wrong directory** for a repo whose skills live at the root (a published-plugin layout) while its rules sit under `.claude/`. Both candidate bases are read now, and that shape has its own test.

Your framing of why this one mattered — "this is exactly the layer where the hardest repo policies live, and to a harness audit they do not exist" — is quoted in the field's doc comment.

---
_Generated by [Claude Code](https://claude.ai/code)_
